### ISSUE TITLE
Fix cobertura export for VSTS

### ISSUE THREAD

**Issue by amethystwu** [OpenCppCoverage/OpenCppCoverage#22]
Thanks for great efforts on this nice tool.

May I know when "[Fix cobertura export for Visual Studio Team Services](https://opencppcoverage.codeplex.com/discussions/660773)" mentioned in the road map could be accomplished and released? I do not find any related commits for it and currently the content of code coverage section in our build dashboard is still wrong due to this.

Thanks in advance.

**Comment by OpenCppCoverage** [comment]
Hello,

Thank you for remind me this issue.
I have no knowledge about Visual Studio Team Service. Is there an easy way for me to test it (I only have VS Community Edition)?
 
If not, based on an old CodePlex discussion someone (maybe you) fixed in the following way:  
 
 "fixed" 0.9.5 version:
 ```
 <coverage line-rate="0.40756464685449634" timestamp="1" version="1" branch-rate="1" branches-valid="1" branches-covered="1" complexity="1" lines-valid="2591" lines-covered="1056">
 <package name="D:\\a\\1\\s\\cmake\\Debug\\SharedComponents.Tests.exe" line-rate="0.40756464685449634" complexity="1" branch-rate="1">
 <class name="base64_rfc4648.hpp" filename="a\\1\\s\\cppcodecb64decoderencoder\\cppcodec\\include\\base64_rfc4648.hpp" line-rate="0.9285714285714286" complexity="1" branch-rate="1">
 ```
 
 Normal 0.9.6 version:
 ```
 <coverage line-rate="0.40772200772200773" branch-rate="0" complexity="0" branches-covered="0" branches-valid="0" timestamp="0" lines-covered="0" lines-valid="0" version="0">
 <package name="D:\\a\\1\\s\\cmake\\Debug\\SharedComponents.Tests.exe" line-rate="0.40772200772200773" branch-rate="0" complexity="0">
 <class name="base64_rfc4648.hpp" filename="a\\1\\s\\cppcodecb64decoderencoder\\cppcodec\\include\\base64_rfc4648.hpp" line-rate="0.9285714285714286" branch-rate="0" complexity="0"> 
 ```
 
There are many changes but I would like to change only what was required (branches-valid="1" seems strange for example). Can you perform a small test and let me know which changes are really needed?
  
OpenCppCoverage

**Comment by amethystwu** [comment]
Hello,

Thanks for addressing this issue promptly. The VSTS uses **line-valid** divided by **line-covered** to calculate the coverage percentage (I have no idea why it doesn't use **line-rate** directly). Currently  both attributes are always 0 therefore it always display 100% on the VSTS dashboard. And we do not care about the branch rate as of now.

As for VSTS, you can play with it with free account online: https://www.visualstudio.com/team-services/. We can also test your new release on our build server if necessary.

Thanks & Cheers

**Comment by OpenCppCoverage** [comment]
Hello,

Please let me know if it solves the problem. 
Note: This is an alpha version and some features may not work.

OpenCppCoverage
[OpenCppCoverageSetup-x64-0.9.6.2-alpha.exe.zip](https://github.com/OpenCppCoverage/OpenCppCoverage/files/1721844/OpenCppCoverageSetup-x64-0.9.6.2-alpha.exe.zip)

**Comment by amethystwu** [comment]
Note. Will update you as soon as we have tested it.

Thanks & Cheers

**Comment by amethystwu** [comment]
Update: the fix has been verified. Thanks for the efforts.

Cheers.

**Comment by OpenCppCoverage** [comment]
Hello,

The fix is available in [OpenCppCoverage 0.9.7.0](https://github.com/OpenCppCoverage/OpenCppCoverage/releases/tag/release-0.9.7.0).

OpenCppCoverage

**Comment by ericlemes** [comment]
I'm trying to publish results using OpenCppCoverage 0.9.7.0 with Visual Studio Online and having issues...

Any guesses?

2018-03-15T17:22:42.9607744Z Reading code coverage summary from 'Scripts\vstest.consoleCoverage.xml'
2018-03-15T17:22:42.9747378Z ##[error]Unable to process command '##vso[codecoverage.publish codecoveragetool=Cobertura;summaryfile=Scripts\vstest.consoleCoverage.xml;reportdirectory=Scripts\CoverageReport;]' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
2018-03-15T17:22:42.9761564Z ##[error]File 'Scripts\vstest.consoleCoverage.xml' does not exist or is not accessible.
2018-03-15T17:22:43.3789104Z ##[section]Finishing: Publish code coverage
